[belligerentbeagle] iP

belligerentbeagle opened the pull request:
# Snoopy
> "Woof woof woof, woof woof WOOF woof woof!" - [Snoopy](https://www.youtube.com/watch?v=Kxvpe_s5TkE&ab_channel=dr725)

Snoopy helps you to remember the things you don't!
+ Like this
+ And this by Tuesday
+ And this! from Wednesday to Thursday!

All you need to do _is_:
1. Download it from [here](github.com/belligerentbeagle).
2. Double-click on the download.
3. Add your tasks.
4. Let it manage tasks for you! 😺 

And.... best part of all....  it is **FREE!**! Woof woof woof!

If you want to do it the old-fashioned way... you can as well! Here is the `main` method to get you started:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
} 
```

Features that have yet to be implemented:
- [ ] batch operations
- [x] edit operations
- [ ] ~~Bark~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

belligerentbeagle committed:
Rename, Greet and Exit!

belligerentbeagle committed:
Echo

belligerentbeagle committed:
Optimise greetings

belligerentbeagle committed:
Add List function

belligerentbeagle committed:
Mark as done, Change BingBong to Snoopy

belligerentbeagle committed:
Add Deadline, Event, Todo

belligerentbeagle committed:
Update test cases input and expected

belligerentbeagle committed:
Change Class name back to Duke

belligerentbeagle committed:
Create exception handling

belligerentbeagle committed:
Add delete Task function

belligerentbeagle committed:
Add Enums

belligerentbeagle committed:
Read from file

belligerentbeagle committed:
Merge branch 'branch-Level-7'

belligerentbeagle committed:
improve Date parsing

belligerentbeagle committed:
Merge branch 'branch-Level-8'

belligerentbeagle committed:
Split files into better OOP

belligerentbeagle committed:
Merge branch 'branch-A-MoreOOP'

belligerentbeagle committed:
Organise into packages

belligerentbeagle committed:
Merge branch 'branch-A-Packages'

belligerentbeagle committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

belligerentbeagle committed:
Added JUnit tests

belligerentbeagle committed:
make jar file

belligerentbeagle committed:
Update with javadoc for most public functions

belligerentbeagle committed:
Merge branch 'branch-A-JavaDoc'

belligerentbeagle committed:
Update Java into format

belligerentbeagle committed:
Improve coding standard

belligerentbeagle committed:
Merge branch 'branch-A-CodingStandard'

belligerentbeagle committed:
Add Search function

belligerentbeagle committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java

belligerentbeagle committed:
Improve Java coding standard

itstrueitstrueitsrealitsreal commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,270 @@
+     * @param isVerbose helps to ensure whether we are preloading (includes the need to save the entries)
+     * @throws RuntimeException
+     */
+    public static void processCommand(String userInput, TaskList todos, Boolean isVerbose) throws RuntimeException {
```
This is a massive method (line 39 - 218)! You could consider using a separate utility class to read and parse input, and separate methods to return the desired actions.

itstrueitstrueitsrealitsreal commented on src/main/java/exceptions/TaskNotExistException.java:
```diff
@@ -0,0 +1,18 @@
+package exceptions;
+
+public class TaskNotExistException extends DukeException {
```
You could consider renaming this MissingTaskException, which would be shorter and more grammatically correct, but this is more of a nitpick.

belligerentbeagle replied:
Thank you for your valued recommendation, i will get to implementing that! Happy Chinese New Year!

itstrueitstrueitsrealitsreal commented on src/main/java/service/Storage.java:
```diff
@@ -0,0 +1,146 @@
+
+public class Storage {
+
+    //private service.TaskList taskList;
```
Can consider removing commented out lines of code.

itstrueitstrueitsrealitsreal commented on src/main/java/service/Storage.java:
```diff
@@ -0,0 +1,146 @@
+                break;
+            }
+
+            if (done.equals("1")) { // done
```
Can consider standardising your comment layout, either above code or in the same line as code.

itstrueitstrueitsrealitsreal commented on src/test/java/model/DeadlineTest.java:
```diff
@@ -0,0 +1,23 @@
+public class DeadlineTest {
+
+    @Test
+    public void dateParsing_properDate_dateParsedCorrectlyForSaving(){
```
You could consider naming the test methods following the coding standards for naming test methods.

itstrueitstrueitsrealitsreal commented:
$`Woof`$ $`Woof`$

belligerentbeagle committed:
Add javafx partial

belligerentbeagle committed:
Merge branch 'master' into branch-Level-10

belligerentbeagle committed:
Update GUI with fxml loaders

belligerentbeagle committed:
Merge branch 'branch-Level-10'

belligerentbeagle committed:
Clean up code

belligerentbeagle committed:
Add assert conditions

belligerentbeagle committed:
Improve code quality

belligerentbeagle committed:
Merge pull request #1 from belligerentbeagle/branch-A-Assertions

Add assert conditions

belligerentbeagle committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assert conditions

belligerentbeagle committed:
Bug fixing and refactoring for code quality

belligerentbeagle committed:
Improve code quality

belligerentbeagle committed:
Improve code quality further.

Added a new class 'TaskManager' to effectively process commands. The entirety of its logic was previously in Duke, which made Duke extremely lengthy. This also allowed for method refactoring, making the code follow SLAP closer.

belligerentbeagle committed:
Merge pull request #2 from belligerentbeagle/branch-A-CodeQuality

Improve code quality

belligerentbeagle committed:
Add ability for tags

belligerentbeagle committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Add ability for tags

belligerentbeagle committed:
Add Snoopy-like responses to command responses

belligerentbeagle committed:
Merge branch 'branch-A-Personality'

* branch-A-Personality:
  Add Snoopy-like responses to command responses

belligerentbeagle committed:
Edit user guide, fix testing

belligerentbeagle committed:
Fix screenshot location

belligerentbeagle committed:
Improve user guide

belligerentbeagle committed:
Fix a ton of bugs and created a new release of snoopy

belligerentbeagle committed:
Improve test cases

belligerentbeagle committed:
Update build gradle so that jar file can be launched in Windows as well

belligerentbeagle committed:
Fix first run app fail bug due to repeated createNewFile

belligerentbeagle committed:
Update README with new UI.png link

belligerentbeagle committed:
Update README.md
